src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -327,49 +327,136 @@ public static SemanticModel GetOriginalSemanticModel(this SemanticModel semantic
         public static ValueUsageInfo GetValueUsageInfo(
             this SemanticModel semanticModel,
             SyntaxNode node,
+            ISyntaxFactsService syntaxFacts,
             ISemanticFactsService semanticFacts,
             CancellationToken cancellationToken)
         {
-            if (semanticFacts.IsInOutContext(semanticModel, node, cancellationToken))
-            {
-                return ValueUsageInfo.WritableReference;
-            }
-            else if (semanticFacts.IsInRefContext(semanticModel, node, cancellationToken))
-            {
-                return ValueUsageInfo.ReadableWritableReference;
-            }
-            else if (semanticFacts.IsInInContext(semanticModel, node, cancellationToken))
-            {
-                return ValueUsageInfo.ReadableReference;
-            }
-            else if (semanticFacts.IsOnlyWrittenTo(semanticModel, node, cancellationToken))
-            {
-                return ValueUsageInfo.Write;
-            }
-            else
+            return syntaxFacts.IsInNamespaceOrTypeContext(node)
+                ? GetNamespaceOrTypeUsageInfo()
+                : GetValueUsageInfo();
```

Comment:
another example of how these feel like different types.  though i'm ok with SymbolUsageInfo